qt: Ensure that ModalOverlay is resized properly

hebasto opened the pull request:
In certain circumstances the `ModalOverlay` widget is not sized properly:
- #17269
- #17967 
- https://github.com/bitcoin/bitcoin/pull/17968#pullrequestreview-350753107

On master (f018d0c9cd7f408dac016b6bfc873670de713d27) this bug looks like this:
![DeepinScreenshot_bitcoin-qt_20200120193402](https://user-images.githubusercontent.com/32963518/72748165-298b2a80-3bbf-11ea-810d-2966f08e496a.png)

With this PR the wallet frame looks ok:
![DeepinScreenshot_bitcoin-qt_20200120195241](https://user-images.githubusercontent.com/32963518/72748388-c64dc800-3bbf-11ea-8875-1ba1899b3513.png)

Fix #17269
Fix #17967

emilengler commented:
utACK f3bac3a1874313796cf2d7acea2d3da74f36a2b0

promag reviewed (commented):
Unrelated but while we are here you can notice that if the window height is increased while the modal is hiding then it will stay a little visible like:

<img width="949" alt="Screenshot 2020-01-30 at 11 20 42" src="https://user-images.githubusercontent.com/3534524/73445461-9e462d80-4352-11ea-9331-cf45e3cb13a0.png">

because the animation end value doesn't track the new height.

Honestly why not drop the animation and eventually make it a modal dialog instead of being embedded?

hebasto commented:
@promag 
> Unrelated but while we are here you can notice that if the window height is increased while the modal is hiding then it will stay a little visible...

With this PR?

promag commented:
@hebasto here and master. You can check more easily by increasing the animation duration.

hebasto committed:
qt: Track QEvent::Resize during animation

hebasto commented:
Reworked and rebased.

@promag 
> Unrelated but while we are here you can notice that if the window height is increased while the modal is hiding then it will stay a little visible like:
> ...
> because the animation end value doesn't track the new height.

Fixed.

promag commented on src/qt/modaloverlay.h:
```diff
@@ -45,6 +46,7 @@ public Q_SLOTS:
     QVector<QPair<qint64, double> > blockProcessTime;
     bool layerIsVisible;
     bool userClosed;
+    QPropertyAnimation animation{};
```
nit, m_animation, m_slide_animation or something like that. Also, could drop `{}`?

hebasto replied:
Sure! Going to fix.

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/17968#issuecomment-621398261).

promag commented on src/qt/modaloverlay.cpp:
```diff
@@ -48,6 +53,9 @@ bool ModalOverlay::eventFilter(QObject * obj, QEvent * ev) {
             if (!layerIsVisible)
                 setGeometry(0, height(), width(), height());
 
+            if (animation.state() != QAbstractAnimation::Stopped && animation.endValue().toPoint().y()) {
```
nit, `== QPropertyAnimation::Running`? Also add `> 0` or maybe a comment giving some context.

hebasto replied:
According to the [`enum QAbstractAnimation::State`](https://doc.qt.io/qt-5/qabstractanimation.html#State-enum) docs, `state != QAbstractAnimation::Stopped` does not imply `state == QAbstractAnimation::Running` :)

Going to add `>0`.

promag replied:
We don't pause right? Oh well, just drop it and always update when > 0 🙈

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/17968#issuecomment-621398261).

promag reviewed (commented):
Code review ACK 14f68ecdc22e96011e53b6aea5672c853e2c648b.

hebasto commented:
Updated 14f68ecdc22e96011e53b6aea5672c853e2c648b -> 4fc1df41d570ab631a8b47e4427a0b84305e37d1 ([pr17968.02](https://github.com/hebasto/bitcoin/commits/pr17968.02) -> [pr17968.03](https://github.com/hebasto/bitcoin/commits/pr17968.03), [diff](https://github.com/hebasto/bitcoin/compare/pr17968.02..pr17968.03)):

- addressed @promag's comments
- added comment for clarity

promag commented:
Code review ACK 4fc1df41d570ab631a8b47e4427a0b84305e37d1.

jonasschnelli commented:
utACK 4fc1df41d570ab631a8b47e4427a0b84305e37d1
